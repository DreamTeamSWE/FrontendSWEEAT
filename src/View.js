import React from "react";
import Login from "./components/auth/Login/Login.js";
import Benvenuto from "./components/auth/Welcome.js";
import Successo from "./components/auth/Success.js";
import Home from "./components/home/Home.js";
import UserArea from "./components/userArea/UserArea";
import Results from "./components/results/Results.js";
import Navbar from "./components/navbar/Navbar";
import Details from "./components/details/Details";
import Registrazione from "./components/auth/Registration/Registration";
import PWDimenticata from "./components/auth/ForgotPassword/ForgotPassword";
import PWVerifica from "./components/auth/ForgotPassword/ForgotPasswordVerification";
import RootStore from "./stores/RootStore";
import { ContextProvider } from "./Context";
import {
  BrowserRouter as Router,
  Routes,
  Route,
  // Navigate,
} from "react-router-dom";

function View() {
  const store = new RootStore();
  return (
    <ContextProvider value={store}>
      <Router>
        <Navbar />
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/login" element={<Login />} />
          <Route path="/passwordDimenticata" element={<PWDimenticata />} />
          <Route
            path="/passwordDimenticata/verifica"
            element={<PWVerifica />}
          />
          <Route path="/registrazione" element={<Registrazione />} />
          <Route path="/areaPersonale" element={<UserArea />} />
          <Route path="/classifica" element={<Results />} />
          <Route path="/classifica/:locale" element={<Results />} />
          <Route path="/dettagliLocale" element={<Details />} />
          <Route path="/benvenuto" element={<Benvenuto />} />
          <Route path="/successo" element={<Successo />} />
        </Routes>
      </Router>
    </ContextProvider>
  );
}

export default View;
