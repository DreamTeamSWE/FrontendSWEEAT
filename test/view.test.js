import View from '../src/View';
import '@testing-library/jest-dom';
import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import {render, screen} from '@testing-library/react';
import {BrowserRouter as Router} from 'react-router-dom';

configure({adapter: new Adapter()});

describe('Check render View', () => {
  beforeEach(() =>{
  render(
    <View/>
  )});

  test('Check render view', () =>{
    const view = shallow(<View/>);
    expect(screen.getByRole("button", {name: "Cerca"})).toBeInTheDocument();
    expect(view).toMatchSnapshot();;
  })

});
