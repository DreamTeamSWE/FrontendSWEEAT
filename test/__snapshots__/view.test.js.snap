// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check render View Check render view 1`] = `ShallowWrapper {}`;
